plugin/src/main/groovy/com/novoda/buildproperties/internal/DefaultEntriesFactory.groovy

```diff
@@ -0,0 +1,34 @@
+package com.novoda.buildproperties.internal
+
+import com.novoda.buildproperties.BuildProperties
+import com.novoda.buildproperties.Entries
+import com.novoda.buildproperties.ExceptionFactory
+import org.gradle.api.GradleException
+
+class DefaultEntriesFactory implements Entries.Factory {
+
+    private final ExceptionFactory exceptionFactory
+
+    DefaultEntriesFactory(ExceptionFactory exceptionFactory) {
+        this.exceptionFactory = exceptionFactory
+    }
+
+    @Override
+    Entries from(def source) {
+        if (source instanceof BuildProperties) {
```

Comment:
We might consider casing here. http://mrhaki.blogspot.de/2009/08/groovy-goodness-switch-statement.html

Reply:
TIL, pt2 🙇